src/js/components/AppPageComponent.jsx

```diff
@@ -343,11 +343,8 @@ var AppPageComponent = React.createClass({
     var volumes = AppsStore.getVolumes(model.id);
 
     var tabs = state.tabs.filter(tab =>
-      tab.text !== "Volumes" || (
-        model.container != null &&
-        model.container.volumes != null &&
-        model.container.volumes.length > 0
-      ));
```

Comment:
My reason behind adding that if condition was that at that moment there wasn't the getVolumes function. Your change is very good and makes total sense. 👍 

But I disagree that we do not want to show the volumes tab if no task is running. With local persistent volumes the tasks which are not running are also present in the json without a `task.status` and the volumes are still present on the agent where the offers are reserved since those apps are also resident apps.

So if a task is not running (has no status) the volume it is shown as detached, this logic is implemented in the getVolumes function [AppsStore.js](https://github.com/mesosphere/marathon-ui/blob/master/src/js/stores/AppsStore.js#L350). 

In conclusion the new if statement is very good because it is less complex, but the assumption that we do not want to show the volumes tab if no task is running is wrong. We do not want to show the volumes tab if no volume is present.